### ISSUE TITLE
Batch rename missing?

### ISSUE THREAD

**Issue by otykier** [TabularEditor/TabularEditor3#130]
### Discussed in https://github.com/TabularEditor/TabularEditor3/discussions/129

<div type='discussions-op-text'>

<sup>Originally posted by **ManuelMendozaLevo** July  9, 2021</sup>
Hi all. I´m loving this tool! but I can't find the batch rename option which was previously available on tabular editor 2 (F2 or right click and batch rename).

Am I doing something wrong or it's actually missing from this build.
</div>

**Comment by otykier** [comment]
TE3 does not yet support batch rename through the UI (it’s on the backlog and will be coming soon).

In the mean time, you can use the following C# script to replace the word “abc” with “xyz” in all selected columns:

```csharp
// Replaces all occurences of the string “abc” with “xyz” for all currently selected columns in the TOM Explorer:
Selected.Columns.Rename("abc", "xyz");
```

The Rename method has [this signature](https://github.com/TabularEditor/TabularEditor/blob/885ff9f84f6497b7958a402854d959d747e0e0a2/TabularEditor/UI/UISelectionList.cs#L238) (it works both in TE2 and TE3), and as you can see, you can set the 3rd argument to “true” to support RegEx search and replace.

Unfortunately, the method only works for the current selection in the TOM Explorer (Selected.Columns, Selected.Measures, etc.) as in the example above. If you need to rename columns across the entire model, regardless of the selection in the TOM Explorer, you would have write the replace logic manually:

```csharp
// Replaces all occurences of the string “abc” with “xyz” in all column names across all tables in the model:
foreach(var column in Model.AllColumns)
{
    column.Name = column.Name.Replace("abc", "xyz");
}
```

Of course in this latter example, you can iterate over any collection of named objects.

**Comment by otykier** [comment]
Batch rename is now available in [3.1.0](https://github.com/TabularEditor/TabularEditor3/releases/tag/3.1.0).
